src/declarativetreeviewextension_p.h

```diff
@@ -45,22 +46,27 @@ class DeclarativeTreeViewExtension : public DeclarativeItemViewExtension
   // repeat property declarations, qmlRegisterExtendedType doesn't see the ones from base class
   Q_PROPERTY(QAbstractItemModel* model READ model WRITE setModel NOTIFY modelChanged)
   Q_PROPERTY(QItemSelectionModel* selectionModel READ selectionModel WRITE setSelectionModel NOTIFY selectionModelChanged)
+  Q_PROPERTY(QAbstractItemDelegate* itemDelegate READ itemDelegate WRITE setItemDelegate NOTIFY itemDelegateChanged)
 
   Q_PROPERTY(QQmlListProperty<QObject> data READ data DESIGNABLE false)
 
   Q_CLASSINFO("DefaultProperty", "data")
 
   public:
-    explicit DeclarativeTreeViewExtension(QObject *parent = 0);
+    explicit DeclarativeTreeViewExtension(QObject *parent = Q_NULLPTR);
```

Comment:
In general a valid change, but not related to support for delegates.
Such a cleanup should be in a separate commit and probably across classes.